src/Common/Core/Header/Cookie.php

```diff
@@ -0,0 +1,180 @@
+                $raw,
+                $sameSite
+            )
+        );
+    }
+
+    public function setCookie(SymfonyCookie $cookie): void
+    {
+        $this->newCookiesHeaderBag->setCookie($cookie);
+    }
+
+    /**
+     * Deletes a cookie
+     *
+     * @param string $name
+     * @param string $path
+     * @param string|null $domain
+     * @param bool|null $secure
+     * @param bool $httpOnly
+     */
+    public function delete(
+        string $name,
+        string $path = '/',
+        string $domain = null,
+        bool $secure = null,
+        bool $httpOnly = true
+    ): void {
+        // clear in the browser
+        $this->newCookiesHeaderBag->clearCookie(
+            $name,
+            $path,
+            $this->normalizeDomain($domain),
+            $this->normalizeSecure($secure),
+            $httpOnly
+        );
+
+        // remove from the new cookies array
+        $this->newCookiesHeaderBag->removeCookie($name, $path, $this->normalizeDomain($domain));
+
+        // remove from the request
+        unset($_COOKIE[$name]);
+        Model::getRequest()->cookies->remove($name);
+    }
+
+    public function get(string $name, string $default = null): ?string
+    {
+        return Model::getRequest()->cookies->get($name, $default);
+    }
+
+    public function has(string $name): bool
+    {
+        return Model::getRequest()->cookies->has($name);
+    }
+
+    public function all(): array
+    {
+        return Model::getRequest()->cookies->all();
+    }
+
+    /**
+     * This will set the secure flag to true if no specific secure flag is given and we are on https
+     *
+     * @param bool|null $secure
+     *
+     * @return bool
+     */
+    private function normalizeSecure(bool $secure = null): bool
+    {
+        if ($secure !== null) {
+            return $secure;
+        }
+
+        /*
+         detect if we are using HTTPS, this wil only work in Apache, if you are using nginx you should add the
+         code below into your config:
+             ssl on;
+            fastcgi_param HTTPS on;
+
+         for lighttpd you should add:
+             setenv.add-environment = ("HTTPS" => "on")
+         */
+
+        return (isset($_SERVER['HTTPS']) && mb_strtolower($_SERVER['HTTPS']) === 'on');
+    }
+
+    /**
+     * This will set the domain to the current host if the domain is null
+     *
+     * @param string|null $domain
+     *
+     * @return string|null
+     */
+    private function normalizeDomain(string $domain = null): ?string
+    {
+        if ($domain !== null) {
+            return $domain;
+        }
+
+        if (Model::requestIsAvailable()) {
+            return '.' . Model::getRequest()->getHost();
+        }
+
+        return null;
+    }
+
+    /**
+     * Has the visitor allowed cookies?
+     *
+     * @return bool
+     */
+    public function hasAllowedCookies(): bool
+    {
+        return $this->getSerialized('cookie_bar_agree', 'N') === 'N';
```

Comment:
cookies need to contain a string value, booleans are not allowed, thats why I kept the Y and N here